docs/migrate-to-antrea.md

```diff
@@ -0,0 +1,69 @@
+# Migrate from other CNI to Antrea
+
+This document describes how to migrate from other CNIs to Antrea.
+
+NOTE: The following is a reference list of CNIs and versions we have already
+verified the migration process. CNIs and versions that are not listed here
+might also work, please create an issue if you run into problems during the
+migration to Antrea. In addition, during the migration process, no Kubernetes
+resources should be created or deleted, otherwise the migration process might
+fail or some unexpected problems might occur.
+
+| CNI     | Version |
+|---------|---------|
+| Calico  | v3.26   |
+| Flannel | v0.22.0 |
+
+With the help of migration [script](../hack/migrate-to-antrea.sh),
+the migration process is fully automated. You can use the following
+command to start the migration:
+
+```bash
+./hack/migrate-to-antrea.sh
+```
+
+More details about the migration process are described in the following
+sections. And you can learn how to use the script by running
+`./hack/migrate-to-antrea.sh --help`.
+
+The migration process is divided into two steps:
+
+1. Install Antrea in the cluster.
+2. Restart all Pods in the cluster in-place.
+
+## Install Antrea
+
+After the requirement check, the script will install Antrea in the cluster
+by the following command:
+
+```bash
+kubectl apply -f ${ANTREA_MANIFEST} --kubeconfig $KUBECONFIG
+```
+
+## Restart Pods
+
+After Antrea is installed in the cluster, the script will restart all Pods in
+the cluster in-place by deploying a DaemonSet named `antrea-migrator`,
+which will run a Pod on each Node. The `antrea-migrator` Pod will all containerd
+tasks of Pods with CNI network on each Node, and the containerd tasks will
+be restarted by the containerd service. In this way, all Pods in the cluster
+will not be rescheduled but will be restarted in-place.
+
+The restart result is will be similar as follows, every Pod's `RESTARTS` will
+be increased by 1:
+
+```bash
+$ kubectl get pod -o wide
+NAME                               READY   STATUS    RESTARTS      AGE    IP          NODE          NOMINATED NODE   READINESS GATES
+migrate-example-6d6b97f96b-29qbq   1/1     Running   1 (24s ago)   2m5s   10.10.1.3   test-worker   <none>           <none>
+migrate-example-6d6b97f96b-dqx2g   1/1     Running   1 (23s ago)   2m5s   10.10.1.6   test-worker   <none>           <none>
+migrate-example-6d6b97f96b-jpflg   1/1     Running   1 (23s ago)   2m5s   10.10.1.5   test-worker   <none>           <none>
+```
+
+## Clean up old CNI
+
+After all Pods are restarted, Antrea will take over the network management and
+IPAM, but the old CNI may still be running in the cluster. There might be
+static routes or iptable rules conflicts between Antrea and the old CNI. So you
+need to clean up the old CNI manually after the migration is finished.
```

Comment:
I thought the old CNI should be cleaned up before the migration, no? otherwise how the migration script deals with race condition, and how to ensure the old CNI doesn't create its resource back?

The process in my mind is:
1. User uninstalls old CNI (users know how to uninstall it much better than us)
2. Install Antrea in whatever way they like (As explained above, this is not a difficult work, no need to make the migration step specific to one installation method)
3. Run the migration script.

Reply:
This will make switching CNI in a managed cluster easier, as the plugin manager just needs to removes the old one, installs new one, run a post-deploy script.